### PR TITLE
chore: use the latest cometbft-db in v0.38.x

### PR DESCRIPTION
**Pull request by faddat**
The goal of this PR is to allow users to use the latest set of db backends with the 38 release series.  Alternatively, we could add something to the cometbft-db v0.13.0 release notes that let users of cometbft-db know that it and following versions are incompatible with v0.38.x.

* https://github.com/cometbft/cometbft/issues/4295

---

#### PR checklist

- [ ] Tests written/updated
- [ ] Changelog entry added in `.changelog` (we use [unclog](https://github.com/informalsystems/unclog) to manage our changelog)
- [ ] Updated relevant documentation (`docs/` or `spec/`) and code comments

### CODE REVIEW

**Comment by faddat** [comment]
oh hey thank you!  I noticed that this landed in the latest release, really appreciate it :)

Should I close this PR now?

**Comment by melekes** [comment]
It would be nice to update cometbft-db to v0.15.0 and Go to 1.23 here. Would you mind updating the PR? If not, I can do it myself, too. Thanks 👍

**Comment by faddat** [comment]
Yeah I'm happy to update the PR 🙂

**Comment by faddat** [comment]
Yeah I'm happy to update the PR 🙂

**Comment by faddat** [comment]
@melekes - I think that I got the merge right but am not 100% sure.

File: go.mod [hunk 1]
```diff
@@ -39,7 +41,7 @@ require github.com/syndtr/goleveldb v1.0.1-0.20210819022825-2ae1ddf74ef7
 
 require (
 	github.com/Masterminds/semver/v3 v3.3.0
-	github.com/cometbft/cometbft-db v0.14.1
+	github.com/cometbft/cometbft-db v1.0.1
```
Comment (melekes) [inline go.mod:44]: I don't think it's okay to bump a major version here

```suggestion
	github.com/cometbft/cometbft-db v0.15.0
```

File: test/e2e/docker/Dockerfile [hunk 2]
```diff
@@ -1,7 +1,7 @@
 # We need to build in a Linux environment to support C libraries, e.g. RocksDB.
 # We use Debian instead of Alpine, so that we can use binary database packages
 # instead of spending time compiling them.
-FROM cometbft/cometbft-db-testing:v0.14.1
+FROM cometbft/cometbft-db-testing:v1.0.1
```
Comment (melekes) [inline test/e2e/docker/Dockerfile:4]:
```suggestion
FROM cometbft/cometbft-db-testing:v0.15.0
```

**Review by melekes — COMMENTED** [review]
:

**Comment by faddat** [comment]
Thanks for the review!

**Comment by melekes** [comment]
You probably need to run `go mod tidy` once again.

**Comment by faddat** [comment]
tidy did it :)

**Review by melekes — APPROVED** [review]
👍

**Pull request merged**